services/sinner/contracts/origin/parser.go

```diff
@@ -0,0 +1,164 @@
+// Package origin is the origin contract parser.
+package origin
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/synapsecns/sanguine/services/sinner/logger"
+
+	"github.com/synapsecns/sanguine/agents/contracts/origin"
+	"github.com/synapsecns/sanguine/agents/types"
+	"github.com/synapsecns/sanguine/services/sinner/db"
+	"github.com/synapsecns/sanguine/services/sinner/db/model"
+	sinnerTypes "github.com/synapsecns/sanguine/services/sinner/types"
+)
+
+// ParserImpl is the parser for the origin contract.
+type ParserImpl struct {
+	filterer *origin.OriginFilterer
+	// parser is the parser interface.
+	parser origin.Parser
+	// db is the database
+	db db.EventDB
+	// txMap is a map of tx hashes to tx data. Exported for testing.
+	txMap map[string]sinnerTypes.TxSupplementalInfo
```

Comment:
for each range of logs to process, the parser also has a map of txs so it can get the sender and timestamp data from scribe instead of having to request each time.